### PR TITLE
Adding design principles overview to additional resources

### LINKED ISSUE #165: Should we have a subsection for short mentions and explanations of design principles (SOLID, DRY, ..)
While I was looking through patterns on this repo, I noticed that people does not mention or refer to [SOLID](https://en.wikipedia.org/wiki/SOLID) principles at all. When designing application we usually think in terms of design principles, e.g, DRY, SOLID, etc, and only then we choose specific patterns to achieve our goals. I think it would be, in particular, useful and informative to create a section for design principles too. I mean, sometimes Strategy, Command, etc,...patterns do not mean much. But if I know, for example, that a specific pattern allows me to to stay DRY, then I consider it. Personally for me, software design is built upon principles, and those principles are achieved using concrete patterns. Just my opinion :-)

_Originally posted by @fade2black in https://github.com/rust-unofficial/patterns/pull/146#issuecomment-754631542_

### PR DESCRIPTION
**Pull request by simonsan**
Closes #165

### CODE REVIEW

File: additional_resources.md [hunk 1]
```diff
@@ -11,3 +11,55 @@ A collection of complementary helpful content
...
+- [Liscov Substitution Principle (LSP)](https://en.wikipedia.org/wiki/Liskov_substitution_principle) -
+  "Objects in a program should be replaceable with instances of their subtypes without altering the correctness
+  of that program."
+- [Interface Segregation Principle (ISP)](https://en.wikipedia.org/wiki/Interface_segregation_principle) -
+  "Many client-specific interfaces are better than one general-purpose interface."
+- [Dependency Inversion Principle (DIP)](https://en.wikipedia.org/wiki/Dependency_inversion_principle) -
+  One should "depend upon abstractions, [not] concretions."
+
+### [DRY (Don’t Repeat Yourself)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
+- "Every piece of knowledge must have a single, unambiguous, authoritative representation within a system"
```
Comment (MarcoIeni) [inline additional_resources.md]:
```suggestion
- [DRY (Don’t Repeat Yourself)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself): "Every piece of knowledge must have a single, unambiguous, authoritative representation within a system"
```
I think that like they are nicer to see

File: additional_resources.md [hunk 2]
```diff
@@ -11,3 +11,68 @@ A collection of complementary helpful content
...
+  "Objects in a program should be replaceable with instances of their subtypes without altering the correctness
+  of that program."
+- [Interface Segregation Principle (ISP)](https://en.wikipedia.org/wiki/Interface_segregation_principle) -
+  "Many client-specific interfaces are better than one general-purpose interface."
+- [Dependency Inversion Principle (DIP)](https://en.wikipedia.org/wiki/Dependency_inversion_principle) -
+  One should "depend upon abstractions, [not] concretions."
+
+### [DRY (Don’t Repeat Yourself)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
+
+- "Every piece of knowledge must have a single, unambiguous, authoritative representation within a system"
```
Comment (MarcoIeni) [inline additional_resources.md:34]:
```suggestion
- [DRY (Don’t Repeat Yourself)](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself): "Every piece of knowledge must have a single, unambiguous, authoritative representation within a system"
```
I think that like this they are nicer to read

Comment (simonsan) [reply additional_resources.md:34]: `:` gets removed by mdlint

Comment (simonsan) [reply additional_resources.md:34]: but i think it's not necessary to add a lint for this

Comment (MarcoIeni) [reply additional_resources.md]: You can use ` - ` or something like that while we investigate this. Anyway, do you like this as a list better?

Comment (simonsan) [reply additional_resources.md]: I think as it's now without hyphen or - it's fine, take a look

**Comment by MarcoIeni** [comment]
This is quite huge, do you like the idea of moving this into a dedicated page under the "Additional Resources" section?

**Comment by simonsan** [comment]
> 
> 
> This is quite huge, do you like the idea of moving this into a dedicated page under the "Additional Resources" section?

can you elaborate? like a `design-principles.md` you mean?

**Comment by MarcoIeni** [comment]
yes!

**Comment by simonsan** [comment]
> 
> 
> yes!

good idea, moved it out

**Comment by MarcoIeni** [comment]
I feel like this shouldn't be an actual chapter of the book. Maybe it could be under "additional resources" or maybe an "appendix".

**Comment by simonsan** [comment]
LGMT?

**Review by MarcoIeni — APPROVED** [review]

**Pull request merged**